test/test_hdf5jl_integration.jl

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env julia
+
+using Test
+using SimpleHDF5
+using HDF5
+
+@testset "HDF5.jl Integration Tests" begin
+
+    @testset "SimpleHDF5 -> HDF5.jl" begin
+        # Create a file with SimpleHDF5
+        tmpdir = mktempdir()
+        test_file = joinpath(tmpdir, "simple_to_hdf5jl.h5")
+
+        # Create test data
+        test_array_1d = [1, 2, 3, 4, 5]
+        test_array_2d = reshape([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12], 3, 4)
+        test_array_3d = reshape(collect(1:24), 2, 3, 4)
```

Comment:
Probably good to add an intentionally-broken `Vector{Bool}` test that can be enabled when https://github.com/JuliaIO/HDF5.jl/issues/1148 is fixed